src/storage/StorageServer.cpp

```diff
@@ -112,7 +112,7 @@ bool StorageServer::start() {
     metaClient_ = std::make_unique<meta::MetaClient>(ioThreadPool_,
                                                      metaAddrs_,
                                                      options);
-    if (!metaClient_->waitForMetadReady()) {
+    if (!metaClient_->waitForMetadReady(FLAGS_meta_client_retry_times)) {
```

Comment:
If pull message failed after some reties, we should ensure the storaged exit, otherwise it has risk to loss data.

Reply:
If the meta retry fails, then waitForMetadReady will return false, where the storage will exit.

Reply:
If we modify in this way, we need to start meta first. Why not stop the `MetaClient`?

Reply:
The default wait time here (10 seconds) I think should also be enough for meta to start?

Reply:
I am curious about why meta-client can't exit the loop after we stop it?